amivapi/studydocs/model.py

```diff
@@ -33,9 +33,51 @@
   user id in the `uploader` field).
 
 - **Admins** can modify items for all users.
+
+<br />
+
+## Summary
+
+For more efficient searching of study documents, a *summary* of available
+metadata fields can be requested with the `summary` keyword in the query string:
+
+```
+/studydocuments?summary
+/studydocuments?where={some filter}&summary
+```
+
+When a summary is requested, the response will contain a new field
+`_summary`, which contains additional information on metadata fields
+listing all distinct values for each field along with the number of documents
+available.
+
+```
+_summary:
+    professor:
+        Prof. Dr. Awesome: 10
+        Prof. Okay: 5
+        ...
+    lecture:
+        ...
+    ...
+```
+
+The summary is only computed for documents matching the current `where` query,
+e.g. when searching for ITET documents, only professors related to ITET
+documents will show up in the summary.
+
+Furthermore, the `author` metadata field is excluded from the summary, as there
```

Comment:
I am against this. I think knowing whose studydoc you are searching is a common usecase. I remember we had a few very popular authors of studydocs, so that is definitely something to search for.
Ignore this if author means uploader, I don't exactly remember how that distinction was. Uploader doesn't matter, but author does.

Reply:
I am not sure. While there might be a handful of authors with multiple documents, the vast majority will probably just have a single document -- won't that just produce a huge summary with lots of values with only a single entry?

But you are right, maybe it is too early to optimize for this.

I can include the author without problems. Should it cause any problems in the future it can easily be disabled.